hw/ip/otbn/dv/tracer/cpp/otbn_trace_listener.h

```diff
@@ -0,0 +1,49 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_HW_OTBN_DV_TRACER_CPP_OTBN_TRACE_LISTENER_H_
+#define OPENTITAN_HW_OTBN_DV_TRACER_CPP_OTBN_TRACE_LISTENER_H_
+
+#include <sstream>
+#include <string>
+#include <vector>
+
+/**
+ * Base class for anything that wants to examine trace output from OTBN. The
+ * simulation that hosts the tracer is responsible for setting up listeners and
+ * routing the DPI `accept_otbn_trace_string` calls to them.
+ */
+class OTBNTraceListener {
+ public:
+  /**
+   * Helper function to split an OTBN trace output up into individual lines.
+   *
+   * @param trace Trace output from OTBN
+   * @return A vector of lines from the trace
+   */
+  static std::vector<std::string> SplitTraceLines(const std::string &trace) {
+    std::stringstream trace_ss(trace);
+    std::string trace_line;
+
+    std::vector<std::string> trace_lines;
+
+    while (std::getline(trace_ss, trace_line, '\n')) {
+      trace_lines.push_back(trace_line);
+    }
```

Comment:
Maybe the efficiency doesn't really matter, but using std::stringstream and std::getline seems a bit overkill here (it does all sorts of clever locale stuff, and we are essentially just doing `strtok`!). Maybe something like this would be more sensible?
```c++
static std::vector<std::string> SplitTraceLines(const std::string &trace) {
  std::vector<std::string> trace_lines;
  size_t pos = 0;
  while (pos != std::string::npos) {
    size_t next_pos = trace.find_first_of('\n', pos);
    trace_lines.push_back(trace.substr(pos, next_pos - pos));
    pos = next_pos;
  }
  return trace_lines;
}
```

Reply:
I prefer the `getline` option. Probably less efficient (though C++ compilers can work fun magic sometimes) but I think the code is cleaner.

Reply:
Hmm, maybe it's just a preference thing.